types/jest/index.d.ts

```diff
@@ -832,8 +887,11 @@ declare namespace jest {
         toHaveReturnedTimes(expected: number): R;
         /**
          * Use to ensure that a mock function returned a specific value.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toHaveReturnedWith(expected: any): R;
+        toHaveReturnedWith<E>(expected: E): R;
```

Comment:
```suggestion
        toHaveReturnedWith<E = any>(expected: E): R;
```

Explicit fallback rather than implicit